src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Codebase\Infrastructure\SourceParser\StructureParser;
+
+use Codebase\Application\Dto\CodebaseSourceDto;
+use Codebase\Application\Dto\SourceParserRequestDto;
+
+class ModuleParser implements StructureParserInterface
+{
+    /**
+     * @var string
+     */
+    protected const CORE_PATH_FORMAT = '%s/*';
+
+    /**
+     * @var string
+     */
+    protected const PROJECT_PATH_FORMAT = '%s/*/*';
+
+    /**
+     * @param \Codebase\Application\Dto\SourceParserRequestDto $codebaseRequestDto
+     * @param \Codebase\Application\Dto\CodebaseSourceDto $codebaseSourceDto
+     *
+     * @return \Codebase\Application\Dto\CodebaseSourceDto
+     */
+    public function parse(SourceParserRequestDto $codebaseRequestDto, CodebaseSourceDto $codebaseSourceDto): CodebaseSourceDto
+    {
+        $coreModules = [];
+        foreach ($codebaseRequestDto->getCorePaths() as $corePath) {
+            foreach ((array)glob(sprintf(static::CORE_PATH_FORMAT, $corePath), GLOB_ONLYDIR) as $dir) {
```

Comment:
could you move it to the variable like $directories or etc?
as far as It looks massive